### PR TITLE
PR

### PR DESCRIPTION
**Pull request by manthankumbhar**

### CODE REVIEW

File: functions/api.js [hunk 1]
```diff
@@ -0,0 +1,320 @@
...:r,isPlainObject:c,isUndefined:s,isDate:function(e){return"[object Date]"===t.call(e)},isFile:function(e){return"[object File]"===t.call(e)},isBlob:function(e){return"[object Blob]"===t.call(e)},isFunction:p,isStream:function(e){return r(e)&&p(e.pipe)},isURLSearchParams:function(e){return"undefined"!=typeof URLSearchParams&&e instanceof URLSearchParams},isStandardBrowserEnv:function(){return("undefined"==typeof navigator||"ReactNative"!==navigator.product&&"NativeScript"!==navigator.product&&"NS"!==navigator.product)&&("undefined"!=typeof window&&"undefined"!=typeof document)},forEach:l,merge:function e(){var a={};function i(i,n){c(a[n])&&c(i)?a[n]=e(a[n],i):c(i)?a[n]=e({},i):o(i)?a[n]=i.slice():a[n]=i}for(var n=0,t=arguments.length;n<t;n++)l(arguments[n],i);return a},extend:function(e,a,i){return l(a,(function(a,t){e[t]=i&&"function"==typeof a?n(a,i):a})),e},trim:function(e){return e.replace(/^\s*/,"").replace(/\s*$/,"")},stripBOM:function(e){return 65279===e.charCodeAt(0)&&(e=e.slice(1)),e}}},function(e,a,i){"undefined"!=typeof process&&"renderer"===process.type?e.exports=i(77):e.exports=i(79)},function(e,a){e.exports=require("path")},function(module,exports,__webpack_require__){
```
Comment (kavadithya) [inline functions/api.js:1]: what is this file?

File: package.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "api-practice",
```
Comment (kavadithya) [inline package.json]: rename to what the project does

File: src/api.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+});
+
+router.post("/", async (req, res) => {
+  let reqData = JSON.parse(req.body);
+  await axios({
+    method: "post",
+    url: process.env.DISCORD_WEBHOOK_URL || DISCORD_WEBHOOK_URL,
+    data: {
+      content: JSON.stringify(
+        `${reqData.repository.owner.login} just pushed a commit with message - '${reqData.head_commit.message}' to <${reqData.repository.name}>`
```
Comment (kavadithya) [inline src/api.js]: prepare message outside, store to var, and use it here

Comment (kavadithya) [inline src/api.js]: handle failure cases where the message comes in badly formatted- return 400 in those cases

File: src/api.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const serverless = require("serverless-http");
+const axios = require("axios");
+
+require("dotenv").config();
+
+const app = express();
+
+const router = express.Router();
+
+router.get("/", (req, res) => {
```
Comment (kavadithya) [inline src/api.js]: rename to /ready

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+node_modules
```
Comment (kavadithya) [inline .gitignore:1]: rename repo to project name

**Review by kavadithya — COMMENTED** [review]
added comments

File: functions/api.js (hunk 1 above)
Comment (manthankumbhar) [reply functions/api.js:1]: This file is for netlify hosting, followed the tutorial

File: .gitignore (hunk 5 above)
Comment (manthankumbhar) [reply .gitignore:1]: ok

File: src/api.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+const router = express.Router();
+
+router.get("/ready", (req, res) => {
+  res.json({ message: "hello" });
+});
+
+router.post("/ready", async (req, res, error) => {
+  try {
+    let reqData = JSON.parse(req.body);
+    const message = `${reqData.repository.owner.login} just pushed a commit with message - '${reqData.head_commit.message}' to <${reqData.repository.name}>`;
```
Comment (kavadithya) [inline src/api.js]: null check for params missing - this won't cause error and throw 400

File: src/api.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      method: "post",
+      url: process.env.DISCORD_WEBHOOK_URL || DISCORD_WEBHOOK_URL,
+      data: {
+        content: JSON.stringify(message),
+        ContentType: "application/json",
+      },
+    });
+    res.json({ message: "success" });
+  } catch (error) {
+    res.status(400).json({ error: "bad format" });
```
Comment (kavadithya) [inline src/api.js]: this shoud be unexpected error, 500. 
Missing params should be 400

File: src/api.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+});
+
+router.post("/ready", async (req, res) => {
+  // try {
+  var reqData = JSON.parse(req.body);
+  var message = `${reqData.repository.owner.login} just pushed a commit with message - '${reqData.head_commit.message}' to <${reqData.repository.name}>`;
+  if (
+    reqData.repository.owner.login &&
+    reqData.head_commit.message &&
+    reqData.repository.name === null
```
Comment (kavadithya) [inline src/api.js:20]: should be checking all the values used; 
if login is null here, it will not throw 400;

File: src/api.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+const router = express.Router();
+
+router.get("/ready", (req, res) => {
+  res.json({ message: "hello" });
+});
+
+router.post("/ready", async (req, res) => {
+  // try {
+  var reqData = JSON.parse(req.body);
+  var message = `${reqData.repository.owner.login} just pushed a commit with message - '${reqData.head_commit.message}' to <${reqData.repository.name}>`;
```
Comment (kavadithya) [inline src/api.js]: message should be constructed only after the test below is done.

File: src/api.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  if (
+    reqData.repository.owner.login &&
+    reqData.head_commit.message &&
+    reqData.repository.name === null
+  ) {
+    res.status(400).json({ error: "bad format" });
+  } else {
+    await axios({
+      method: "post",
```
Comment (kavadithya) [inline src/api.js]: post should be caps: POST

File: src/api.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+      url: process.env.DISCORD_WEBHOOK_URL || DISCORD_WEBHOOK_URL,
+      data: {
+        content: JSON.stringify(message),
+        ContentType: "application/json",
+      },
+    });
+    res.json({ message: "success" });
+  }
+  // } catch (error) {
```
Comment (kavadithya) [inline src/api.js]: remove commented code everywhere

File: src/api.js [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+const app = express();
+
+const router = express.Router();
+
+router.get("/ready", (req, res) => {
+  res.json({ message: "hello" });
+});
+
+router.post("/ready", async (req, res) => {
+  // try {
```
Comment (kavadithya) [inline src/api.js]: remvoe commented code everywhere

File: src/api.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+
+router.post("/ready", async (req, res) => {
+  var reqData = JSON.parse(req.body);
+  if (
+    reqData.repository.owner.login === null ||
+    reqData.head_commit.message === null ||
+    reqData.repository.name === null
+  ) {
+    res.status(400).json({ error: "bad format" });
+  } else {
```
Comment (kavadithya) [inline src/api.js]: remove else. 
Return in the previous if block itself.

File: src/api.js [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+});
+
+router.post("/ready", async (req, res) => {
+  var reqData = JSON.parse(req.body);
+  if (
+    reqData.repository.owner.login === null ||
+    reqData.head_commit.message === null ||
+    reqData.repository.name === null
+  ) {
+    res.status(400).json({ error: "bad format" });
```
Comment (kavadithya) [inline src/api.js]: add return statemebt

**Pull request closed without merging**